src/Infrastructure/DependencyInjection.cs

```diff
@@ -37,6 +37,20 @@ public static IServiceCollection AddInfrastructure(this IServiceCollection servi
             services.AddIdentityServer()
                 .AddApiAuthorization<ApplicationUser, ApplicationDbContext>();
 
+            if (configuration.GetValue<bool>("UseInMemoryCache"))
+            {
+                services.AddMemoryCache();
+                services.AddTransient<ICacheService, InMemoryCacheService>();
+            }
+            else if (configuration.GetValue<bool>("UseRedisCache"))
+            {
+                services.AddDistributedRedisCache(options =>
+                {
+                    options.Configuration = "localhost";
+                });
+                services.AddTransient<ICacheService, RedisCacheService>();
+            }
```

Comment:
Wouldn't be better present this `if-else` logic with an extension method like `.AddCaching()` or something else. 

Refactoring caching implementation will not affect this file (Injection.cs)

Reply:
Yes, we can always extract out that to extension method, that we can do later as well! I kept it here because of review, once it gets revised we can always refactor in later stages!
And this is template project, we want to keep as simple and as explicit as possible, then one can do their refactoring as they would like to once they start their own project with it.
I will consider your suggestion and will do refactoring in upcoming commits!